synphot/wrapper.py

```diff
@@ -0,0 +1,153 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""This module is a wrapper for synphot's most commonly used functions.
+
+"""
+
+# STDLIB
+import math
+import warnings
+
+# THIRD-PARTY
+import numpy as np
+
+# ASTROPY
+from astropy import log
+from astropy import units as u
+from astropy.utils.exceptions import (AstropyUserWarning,
+                                      AstropyDeprecationWarning)
+from astropy.modeling.models import Const1D
+
+# LOCAL
+from . import binning, exceptions, units, utils
+from .models import Empirical1D
+from .spectrum import BaseSourceSpectrum, SourceSpectrum, SpectralElement
+from .observation import Observation, ccd_snr, exptime_from_ccd_snr
+
+
+class Observe:
+    """
+    Parameters
+    ----------
+    source_spec : `~synphot.spectrum.SourceSpectrum`
+        The spectrum of the target object. 
+        If distance is given, the spectrum is assumed to be at the source. 
+    distance : `~astropy.units.quantity.Quantity`
+        Distance to the source spectrum. Default is None.
+        If None, source_spec is assumed to be scaled by the distance.
+        If distance is given, the spectrum is assumed to be at the source.
+    diameter : `~astropy.units.quantity.Quantity`
+        Diameter of the telescope aperture.
+    bandpass : str, list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values, or 
+        `~synphot.spectrum.SpectralElement`
+        The bandpass(es) of this observation. Can either be specified by a 
+        string or a list with dimensions 2 x n, where one row is the
+        list of wavelengths associated with the bandpass, and the other
+        row contains the fractional values of the bandpass.
+        Default is a constant model with amplitude = 1, i.e. as if there
+        were no bandpass in effect.
+    quantum_efficiency : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the quantum efficiency of the instrument's CCD. 
+        Default is None.
+    atmospheric_transmission : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the atmopsheric transmission.
+    """
+    @u.quantity_input(diameter=u.m, distance=u.pc, exptime=u.s)
+    def __init__(self, source_spec, diameter, distance=0 * u.pc,
+                 bandpass=SpectralElement(Const1D, amplitude=1),
+                 quantum_efficiency=None, atmospheric_transmission=None,
+                 force=None, exptime=1 * u.s):
```

Comment:
Should things like exptime have a default values? I think that's a thing that will really need to be supplied in every case, right? And distance sticks out as bing an odd one too, nothing you observe is actually at 0 pc.

Reply:
Sure, I agree with `exptime` not being a default param.

As for `distance`, I set the default to 0 instead of something like `False` for a couple of reasons... first as a way to indicate that the user should have already scaled the source spectrum for distance (i.e. `source_spec` should be as it is just before it hits the instrument, i.e. 0 pc), and also because while scaling the spectrum for a star is straight forward ([R / d]^2), I wasn't sure how exactly to do that for something like a galaxy so I put it off for the moment 😂

Reply:
I guess I was envisioning that the user could give the source spectrum as it is at the object, and `Expose` could scale the spectrum given a distance. But on second thought, I think it would be better for all parties if we get rid of `distance` and just specify that `Expose` assumes the user has already scaled the spectrum.